### ISSUE TITLE
Run coverage on Windows as well

### ISSUE THREAD

**Issue by BridgeAR** [nodejs/build#1558]
Our current coverage report at https://coverage.nodejs.org is unix (some linux?) only. It would be great to combine this with the Windows coverage. That way our coverage would be significantly higher and it's easier to spot actual missing parts.

**Comment by mhdawson** [comment]
gcov is currently used and I don't know that it works on windows.  But would be happy if somebody has time to investigate figure out how to get windows coverage added as well. @BridgeAR is that something you are interested in working on?

Having said that, not sure this is a build WG issue...

**Comment by BridgeAR** [comment]
@mhdawson AFAIK gcov is only used for the C++ coverage. It would be great to also know the C++ coverage on Windows but as a first step I would already be happy to include the Windows coverage for JS ([Here](https://coverage.nodejs.org/coverage-d32b5bd567544f5b/root/os.js.html#L124) some [examples](https://coverage.nodejs.org/coverage-d32b5bd567544f5b/root/fs.js.html#L1350) of what should definitely be covered but it's currently not).

I do not know how everything is set up currently but I guess the main issue would be to combine the coverage files from the Windows machine with the Unix ones?

**Comment by mhdawson** [comment]
Combining the files would be one of the challenges.  Does the existing coverage target in the makefile run on Windows?

If you manually ran and combined to get what the new number was that might show what the difference would be.  

In terms of effort/reward it would be interesting to see that number as I think it could be a bit of work to do it.

**Comment by Trott** [comment]
Recent conversation with @bcoe makes me want to revive this. He indicated that it would be no problem to take one coverage results file from Linux, and another coverage results file from a Windows run, and merge them.

I'd also love to see us add coverage for a host compiled without intl since there's code that is not covered otherwise.

**Comment by bcoe** [comment]
:+1: if we could find a way to ship the contents of `out/Release/.coverage` to a shared folder somewhere, the merge is simple.

I'd be happy to help build this tooling; I wonder if we could just throw it up in a cloud storage bucket.

**Comment by Trott** [comment]
I thought we had accomplished this at some point. @bcoe

**Comment by Trott** [comment]
(But looking now, it looks like it's not working now at least.)

**Comment by bcoe** [comment]
👋 we had a regression at some point in our configured CI/CD:

* codecov.io will merge results from a variety of matrices, as  long as the sha matches.
* our coverage implementation works well on Windows.

tldr; I think we just need to carve out the time to reconfigure the appropriate jobs to send output.
